### PR TITLE
validator update

### PR DESCRIPTION
**Pull request by Eugene-hu**
- added loading weights
- fixes moving average
- adds additional features for logging
- reduces the memory of the history

### CODE REVIEW

File: bittensor/_metagraph/__init__.py [hunk 1]
```diff
@@ -124,7 +124,6 @@ def sync ( self, block: Optional[int] = None, lite: bool = True ) -> 'metagraph'
         self.version = torch.nn.Parameter( torch.tensor( [bittensor.__version_as_int__], dtype=torch.int64 ), requires_grad=False )
         self.block = torch.nn.Parameter( torch.tensor( subtensor.block, dtype=torch.int64 ), requires_grad=False )
         self.uids = torch.nn.Parameter( torch.tensor( [ neuron.uid for neuron in self.neurons ], dtype=torch.int64 ), requires_grad=False )
-        self.emission = torch.nn.Parameter( torch.tensor( [ neuron.emission for neuron in self.neurons ], dtype=torch.float32 ), requires_grad=False )
```
Comment (opentaco) [inline bittensor/_metagraph/__init__.py:127]: Why is the emission removed from the metagraph?

File: neurons/text/prompting/validators/core/gating.py [hunk 2]
```diff
@@ -113,7 +113,8 @@ def forward( self, message: str ) -> 'torch.FloatTensor':
                     Scores for each uids as output by the gating model.
         """
         inputs = self.tokenizer( message, return_tensors="pt" ).to( self.device )
-        hidden_states = self.model( **inputs ).last_hidden_state[0, -1, :]
+        with torch.no_grad():
```
Comment (opentaco) [inline neurons/text/prompting/validators/core/gating.py:116]: Would the gating model not need gradients here in order to be trained?

**Review by opentaco — APPROVED** [review]

File: bittensor/_metagraph/__init__.py (hunk 1 above)
Comment (Eugene-hu) [reply bittensor/_metagraph/__init__.py:127]: This was duplicated a bit lower in the metagraph

File: neurons/text/prompting/validators/core/gating.py (hunk 2 above)
Comment (Eugene-hu) [reply neurons/text/prompting/validators/core/gating.py:116]: The idea is that we don't have enough data to fully fine-tune the transformer block. Instead, we should only train the last linear layer and freeze the current state of the transformer.

File: neurons/text/prompting/validators/core/neuron.py [hunk 3]
```diff
@@ -135,6 +134,9 @@ def __init__( self ):
         self.uid = self.wallet.get_uid( subtensor = self.subtensor, netuid = self.config.netuid )
         self.tokenizer = AutoTokenizer.from_pretrained( 'EleutherAI/gpt-j-6b' )
 
+        self.moving_averaged_scores = torch.zeros((self.metagraph.n)).to( self.device )
+        self.alpha = 0.99
```
Comment (isabella618033) [inline neurons/text/prompting/validators/core/neuron.py:138]: alpha = 0.99 seems a bit high, think we used to have alpha = 0.1 (0.9)

File: neurons/text/prompting/validators/core/neuron.py [hunk 4]
```diff
@@ -416,9 +434,12 @@ def train( self ):
                 )
 
                 # Resync metagraph before returning. (sync every 15 min or ~75 blocks)
-                if last_epoch_block % 10 == 0:
+                if self.subtensor.block % 10 == 0:
```
Comment (isabella618033) [inline neurons/text/prompting/validators/core/neuron.py:437]: I think %10 would be syncing every 2 mins, wouldnt that be a bit frequent?

**Review by isabella618033 — APPROVED** [review]
LGTM!

**Review by shibshib — APPROVED** [review]

File: neurons/text/prompting/validators/core/neuron.py (hunk 3 above)
Comment (Eugene-hu) [reply neurons/text/prompting/validators/core/neuron.py:138]: hmm good point, this might need to change as we increase the number of uids. Lets put in a ticket and do deeper analysis once the network is stable
